src/main/java/com/ericsson/gerrit/plugins/eiffel/events/generators/EiffelSourceChangeCreatedEventGenerator.java

```diff
@@ -71,15 +78,17 @@ public static EiffelSourceChangeCreatedEvent generate(PatchSetCreatedEvent patch
         eiffelEvent.eventParams.data.gitIdentifier.branch = branch;
         eiffelEvent.eventParams.data.gitIdentifier.repoName = projectName;
 
-        // TODO
-        // String latestEiffelSourceChangeCreatedEventId =
-        // getLatestEiffelSourceChangeCreatedEventId();
-        // setPreviousVersionLink(latestEiffelSourceChangeCreatedEventId);
-
-        // String latestEiffelSourceChangeSubmittedEventId =
-        // getLatestEiffelSourceChangeSubmittedEventId();
-        // setBaseLink(latestEiffelSourceChangeSubmittedEventId);
+        String previousSourceChangeCreatedEvent = getPreviousEiffelEvent(EventType.SCC_EVENT, projectName, changeId, pluginDirectoryPath);
+        final Link previousVersionLink = createLink(LINK_TYPE_PREVIOUS_VERSION, previousSourceChangeCreatedEvent);
+        if (previousVersionLink != null) {
+            eiffelEvent.eventParams.links.add(previousVersionLink);
+        }
 
+        String previousSourceChangeSubmittedEvent = getPreviousEiffelEvent(EventType.SCS_EVENT, projectName, changeId, pluginDirectoryPath);
+        final Link baseLink = createLink(LINK_TYPE_BASE, previousSourceChangeSubmittedEvent);
```

Comment:
can we move this also to the function createLink?
like createLinkAndAddtoEvent(type, event, eiffelEvent) in that case we can avoid the null check and duplication of code moved to the function.

Reply:
Then that method will do multiple things plus it feels wrong sending in a reference and update it from readability point of view. It doesnt feel as much readable that way.